systemtest/src/test/java/io/strimzi/systemtest/operators/RecoveryIsolatedST.java

```diff
@@ -267,6 +264,24 @@ void testRecoveryFromImpossibleMemoryRequest() {
         KafkaUtils.waitForKafkaReady(sharedClusterName);
     }
 
+    @IsolatedTest
+    @StrimziPodSetTest
+    void testRecoveryFromKafkaAndZookeeperPodDeletion() {
+        final String kafkaName = KafkaResources.kafkaStatefulSetName(sharedClusterName);
+        final String zkName = KafkaResources.zookeeperStatefulSetName(sharedClusterName);
+
+        final LabelSelector kafkaSelector = KafkaResource.getLabelSelector(sharedClusterName, kafkaName);
+        final LabelSelector zkSelector = KafkaResource.getLabelSelector(sharedClusterName, zkName);
+
+        LOGGER.info("Deleting all Kafka and ZK pods");
+        kubeClient(INFRA_NAMESPACE).deletePodsByLabelSelector(kafkaSelector);
+        kubeClient(INFRA_NAMESPACE).deletePodsByLabelSelector(zkSelector);
```

Comment:
I tried that several times on Minikube and OCP, it looked stable to me. But we shouldn't delete all the pods at once if we don't want to loose any data.

I can increase the number of ZK replicas and delete just few pods, instead of all.

Reply:
I increased the number of ZK replicas and in test I'm keeping 1 replica of Kafka and ZK, to not delete them all -> it should be stable enough.